### DISCUSSION TITLE
Looking for "b_" from PyPDF2-1.27 in newer PyPDF2

### DISCUSSION THREAD

**Discussion by pauljohn32** [py-pdf/pypdf#1472] [Q&A] [answered]
I have some old code that imports the text stream and then replaces some characters and writes it out as new PDF.  The code uses a hidden method "b_" to do that.  That's not in the newer PyPDF2 code base.  Can you help with some porting guide info?

We have an import like this now:

```
from PyPDF2.utils import b_
```

And in the content, we are using that to detect the input type and then revise it.

**Comment by MartinThoma** [comment, marked as answer]
You have a good point - we should add that to the migration guide :+1:  Do you want to add it?

https://github.com/py-pdf/PyPDF2/pull/986#issuecomment-1230698069 - this is how you do it

**Reply by pauljohn32** [reply]
I will test the changes, see if they work, and if they do, will circle back to you.  I found a function "b_" in PyPDF2._utils in current release, so am digesting.

**Reply by pauljohn32** [reply]
It can't be as easy as "utils" -> "_utils" :)

**Reply by MartinThoma** [reply]
Huh. Interesting. I was pretty sure I removed it completely :thinking:

**Reply by MartinThoma** [reply]
Ok, let's say it like this: I really recommend to copy-paste that function if you want to keep using it. The `_utils` module should never be used - functions in there might go at any point without warning or change their behavior. Modules starting with an underscore are private.
